### PR TITLE
crimson/tools: integrate perf record and flamegraph to crimson stress tool

### PR DESCRIPTION
**Pull request by aisakaki**

### CODE REVIEW

**Comment by aisakaki** [comment]
@cyx1231st I added perf record and generating flamegraph function to crimson stress tool, could you take a look?

File: tools/crimson/crimson_stress_tool.py [hunk 1]
```diff
@@ -416,6 +418,52 @@ def analyse(self):
...
+            if file ==  "stackcollapse-perf.pl":
+                stackcollapse_perf = True
+            if file == "flamegraph.pl":
+                flamegraph = True
+        if stackcollapse_perf and flamegraph:
+            time.sleep(5)
+            os.system("sudo perf script -i perf.data | ./stackcollapse-perf.pl \
+                --all | ./flamegraph.pl > flamegraph.svg")
+            print("flamegraph generated at current directory.")
+            return ;
```
Comment (cyx1231st) [inline tools/crimson/crimson_stress_tool.py:464]: `;` is unecessary

File: tools/crimson/crimson_stress_tool.py [hunk 2]
```diff
@@ -416,6 +416,52 @@ def analyse(self):
         self.result.close()
         return result_dic
 
+class PerfRecordThread(Task):
+    def __init__(self, env):
+        super().__init__(env)
```
[the remark this answers is not in the document]
Comment (aisakaki) [reply tools/crimson/crimson_stress_tool.py:423]: i set it to run during first 1/4 time to last 1/4 time

File: tools/crimson/crimson_stress_tool.py (hunk 1 above)
Comment (aisakaki) [reply tools/crimson/crimson_stress_tool.py]: all addressed~

File: tools/crimson/crimson_stress_tool.py [hunk 3]
```diff
@@ -416,6 +418,54 @@ def analyse(self):
...
+        # perf begin from first 1/4 time to last 1/4 time
+        self.start_time = round(int(env.args.time) * 0.25)
+        self.last_time = round(int(env.args.time) * 0.75)
```
Comment (cyx1231st) [inline tools/crimson/crimson_stress_tool.py:426]:
```c++
self.last_time = round(int(env.args.time) * 0.5)
```

**Pull request merged**